snapcraft/cli/remote.py

```diff
@@ -0,0 +1,256 @@
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import os
+import uuid
+
+from snapcraft.internal.remote_build import (
+    Worktree,
+    LaunchpadClient,
+    Repo,
+    InfoFile,
+    errors,
+)
+from snapcraft.formatting_utils import humanize_list
+from typing import List, Tuple
+from xdg import BaseDirectory
+from . import echo
+from ._options import get_project
+
+_SUPPORTED_ARCHS = ["amd64", "arm64", "armhf", "i386", "ppc64el", "s390x"]
+
+
+@click.group()
+def remotecli():
+    """Remote build commands"""
+    pass
+
+
+@remotecli.command("remote-build")
+@click.option(
+    "--recover",
+    metavar="<build-number>",
+    type=int,
+    nargs=1,
+    required=False,
+    help="Recover interrupted remote build.",
+)
+@click.option(
+    "--status",
+    metavar="<build-number>",
+    type=int,
+    nargs=1,
+    required=False,
+    help="Display remote build status.",
+)
+@click.option(
+    "--arch",
+    metavar="<arch-list>",
+    type=str,
+    nargs=1,
+    required=False,
+    help="Set architectures to build.",
+)
+@click.option(
+    "--git", is_flag=True, required=False, help="Build a local git repository."
+)
+@click.option(
+    "--user", metavar="<username>", nargs=1, required=False, help="Launchpad username."
+)
+def remote_build(
+    recover: int, status: int, user: str, arch: str, git: bool, echoer=echo
+) -> None:
+    """Dispatch a snap for remote build.
+
+    Command remote-build sends the current project to be built remotely. After the build
+    is complete, packages for each architecture are retrieved and will be available in
+    the local filesystem.
+
+    If the local files are under git revision control, use option --git to build the
+    current branch of this repository. Be aware, however, that untracked files and
+    uncommitted changes will not be used.
+
+    If not specified in the snapcraft.yaml file, the list of architectures to build
+    can be set using the --arch option. If both are specified, the architectures listed
+    in snapcraft.yaml take precedence.
+
+    Interrupted remote builds can be resumed using the --recover option, followed by
+    the build number informed when the remote build was originally dispatched. The
+    current state of the remote build for each architecture can be checked using the
+    --status option.
+
+    \b
+    Examples:
+        snapcraft remote-build
+        snapcraft remote-build --arch=all
+        snapcraft remote-build --git --arch=arm64,armhf
+        snapcraft remote-build --recover 47860738
+        snapcraft remote-build --status 47860738
+    """
+
```

Comment:
can you add a similar warning as we have for promote right here?

Reply:
Warning added.